src/dist/replication/meta_server/meta_server_failure_detector.cpp

```diff
@@ -263,6 +263,10 @@ void meta_server_failure_detector::on_ping(const fd::beacon_msg& beacon, ::dsn::
     {
         ack.is_master = false;
         ack.primary_node = get_primary();
+        {
```

Comment:
@imzhenyu This bug is not so easy to fix.

- First I want to know the exact meaning of "A non-master meta server not handle the Pings." Is it means that a replica server should never receive a FD message with "ack.is_master == true"? If this is what you mean, I think the logic in "replication_failure_detector::end_ping" can be simplified. 

- As what I described above, the replica server's state will be influenced if a non-master meta doesn't handle the Ping. This is mainly because how long a meta server switch to a master depends on the distributed_lock_service's implementation. And it may be longer than the lease period. This implies that a replica server will suffer from being inactive as long as a meta server switching happens.

- Another issue is that no matter how M handles Ping from R, the perfect FD rule will be broken during the period that M has become a master but the first beacon from R hasn't arrived. We may need to store the worker list to the remote storage(e.g. zookeeper) to resolve this.

- Well, the next tricky thing in our project is that the breaking of perfect FD is not so serious. The replica group won't be inconsistent even if two primaries exist, as a preparing message from an old primary will be rejected by the new primary. So I guess we can just keep this bug here for the time being?